Src/FluentAssertions/Primitives/ObjectAssertions.cs

```diff
@@ -251,6 +254,139 @@ public AndConstraint<ObjectAssertions> NotBe(object unexpected, string because =
             return new AndConstraint<ObjectAssertions>(this);
         }
 
+        /// <summary>
+        /// Asserts that an object is enumerable using a foreach loop.
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase explaining why the assertion should be satisfied. If the phrase does not
+        /// start with the word <i>because</i>, it is prepended to the message.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more values to use for filling in any <see cref="string.Format(string,object[])" /> compatible placeholders.
+        /// </param>
+        public AndWhichConstraint<ObjectAssertions, IEnumerable> BeEnumerable(string because = "", params object[] becauseArgs)
+        {
+            AssertIsEnumerable(Subject, because, becauseArgs);
+
+            return new AndWhichConstraint<ObjectAssertions, IEnumerable>(this, new ObjectEnumerable(Subject));
+        }
+
+        public AndConstraint<ObjectAssertions> BeEnumerable(IEnumerable expected, Func<object, object, bool> equalityComparison, string because = "", params object[] becauseArgs)
```

Comment:
Same remark as @jnyrup made: drop the `expected` parameter and return an `AndWhichConstraint` instead